androidTest/java/com/owncloud/android/authentication/SAMLAuthenticatorActivityTest.java

```diff
@@ -0,0 +1,290 @@
+
+    @Rule
+    public ActivityTestRule<AuthenticatorActivity> mActivityRule = new ActivityTestRule<AuthenticatorActivity>(
+            AuthenticatorActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+
+            targetContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext();
+            Intent result = new Intent(targetContext, AuthenticatorActivity.class);
+            result.putExtra(EXTRA_ACTION, AuthenticatorActivity.ACTION_CREATE);
+            result.putExtra(EXTRA_ACCOUNT, "");
+            return result;
+        }
+    };
+
+    @Before
+    public void init() {
+        Bundle arguments = InstrumentationRegistry.getArguments();
+
+        testUser = arguments.getString("TEST_USER");
+        testPassword = arguments.getString("TEST_PASSWORD");
+        testServerURL = arguments.getString("TEST_SERVER_URL");
+        servertype = ServerType.fromValue(Integer.parseInt(arguments.getString("TRUSTED")));
+        webViewUsernameId = arguments.getString("TEST_USERNAME_ID");
+        webViewPasswordId = arguments.getString("TEST_PASSWORD_ID");
+        webViewSubmitXPath = arguments.getString("TEST_SUBMIT_XPATH");
+
+        // UiDevice available form API level 17
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+            /*Point[] coordinates = new Point[4];
+            coordinates[0] = new Point(248, 1020);
+            coordinates[1] = new Point(248, 429);
+            coordinates[2] = new Point(796, 1020);
+            coordinates[3] = new Point(796, 429);*/
+            try {
+                if (!uiDevice.isScreenOn()) {
+                    uiDevice.wakeUp();
+                    //uiDevice.swipe(coordinates, 10);
+                }
+            } catch (RemoteException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    /**
+     *  Login with SAML. Supported with non-secured servers under https
+     */
+    @Test
+    public void test1_check_login_saml()
+            throws InterruptedException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+
+        Log_OC.i(LOG_TAG, "Test Check Login SAML Start");
+
+        SystemClock.sleep(WAIT_INITIAL);
+
+        // Check that login button is disabled
+        onView(withId(R.id.buttonOK)).check(matches(not(isEnabled())));
+
+        onView(withId(R.id.hostUrlInput)).perform(replaceText(testServerURL));
+
+        //Needed to click on the screen to validate the URL
+        onView(withId(R.id.thumbnail)).perform(click());
+
+        //Certificate acceptance in case of non-trusted or expirated
+        if (servertype == ServerType.NON_TRUSTED) {
+
+            SystemClock.sleep(WAIT_CONNECTION);
+            onView(withId(R.id.ok)).perform(click());
+        }
+
+        SystemClock.sleep(WAIT_CONNECTION);
```

Comment:
Sometimes the performance of the tests work faster than the network responses, depending on the device, Android version... so this little delays are included to assure that the assertions are not checked before the actions to avoid fake positives.